code/c++/search_letor.cc

```diff
@@ -0,0 +1,94 @@
+#include <xapian-letor.h>
```

Comment:
This file seems to be heavily based on `xapian/xapian-letor/bin/xapian-rank.cc`, but you've stripped off the copyright statements and licensing boilerplate.  You really must not do that.